src/rgw/store/dbstore/common/dbstore.h

```diff
@@ -1221,6 +1228,11 @@ class DB {
     virtual int FreeDBOps(const DoutPrefixProvider *dpp) { return 0; }
     virtual int InitPrepareParams(const DoutPrefixProvider *dpp, DBOpPrepareParams &params) = 0;
 
+    virtual int beginTransaction(const DoutPrefixProvider *dpp) { return 0; }
+    virtual int endTransaction(const DoutPrefixProvider *dpp) { return 0; }
+    virtual int commitTransaction(const DoutPrefixProvider *dpp) { return 0; }
+    virtual int rollbackTransaction(const DoutPrefixProvider *dpp) { return 0; }
```

Comment:
should these be pure virtual like the others? it doesn't seem like transaction support should be optional for derived classes

Reply:
got it.. thanks